### PR TITLE
[Site] Update student list for 2017

### PR DESCRIPTION
**Pull request by Tzyinc**

### CODE REVIEW

File: TA_Scripts/Update_Students/contactlist.csv [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Name,Email,Faculty,Year,Role,Contact,Blog,Blog Feed,One-liner,Interests
```
Comment (yangshun) [inline TA_Scripts/Update_Students/contactlist.csv]: 😱 We shouldn't commit all their contact details into the repo

File: TA_Scripts/Update_Students/createNewYml.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  node createNewYml.js
+
+  it will generate a new students.yml in the same folder for you to manually update. (instead of replacing the original file)
+  in order to follow existing standards, it will also generate the image id in lowercase and hyphens. The images will have to be renamed manually
+
+  Disclaimer: file follows the Javascript Standard Style, so if you don't like the lack of semi-colons, too bad!
+*/
+
+// generate csv from google docs and put in the same folder, update this name
+var csvFilename = 'contactlist.csv'
```
Comment (yangshun) [inline TA_Scripts/Update_Students/createNewYml.js:13]: We should remove the CSV file and just add the headers here as reference for future batches to use.

File: TA_Scripts/Update_Students/createNewYml.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+var obj = {
+  filename: csvFilename
+}
+
+var callback = function (err, json) {
+  if (err) {
+    return console.error(err)
+  }
+  // console.log(json)
```
Comment (yangshun) [inline TA_Scripts/Update_Students/createNewYml.js]: Remove console.logs

File: TA_Scripts/Update_Students/createNewYml.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  json.map(function (value) {
```
Comment (yangshun) [inline TA_Scripts/Update_Students/createNewYml.js]: You're using map here but not returning anything (you're mutating the value instead). Perhaps you should be using `.forEach`?

File: TA_Scripts/Update_Students/createNewYml.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    value.id = value[nameKey].toLowerCase().replaceAll(' ', '-')
```
Comment (yangshun) [inline TA_Scripts/Update_Students/createNewYml.js]: This operation is called slugify. Doing lower case and replacing spaces with hyphen won't cover all possible cases that might be present in future batches. Would recommend to use a node package to do the slugification instead.

File: TA_Scripts/Update_Students/createNewYml.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  })
+
+  fs.readFile(originalFileDir, function (err, data) {
+    if (err) {
+      return console.error(err)
+    }
+    var oldStr = data.toString()
+    // console.log(oldStr)
```
Comment (yangshun) [inline TA_Scripts/Update_Students/createNewYml.js]: Remove console.log

File: TA_Scripts/Update_Students/createNewYml.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    var currDate = new Date()
+    var thisYear = 1900 + currDate.getYear()
+    var newStr = '- batch:\n  year: ' + thisYear + '\n  students:'
+    for (let student of json) {
+      newStr += '\n  - name: ' + student[nameKey]
```
Comment (yangshun) [inline TA_Scripts/Update_Students/createNewYml.js]: Can use ES6 template strings instead

File: _data/students.yml [hunk 8]
```diff
@@ -1,4 +1,177 @@
...
+    one_liner: ""I'm a person who is always lost, always exploring and always discovering.""
+  - name: Kushagra Goyal
+    id: kushagra-goyal
+    faculty: Computing
+    blog: https://kush3216.blogspot.com/
+    one_liner: "Never say die"
+  - name: Lau Shi Jie
+    id: lau-shi-jie
+    faculty: Business n fass
+    blog: 	 http://cs3216shijie.blogspot.com/
```
Comment (yangshun) [inline _data/students.yml]: Weird spacing here. Maybe the strings should be trimmed

File: TA_Scripts/Update_Students/createNewYml.js (hunk 7 above)
Comment (yangshun) [reply TA_Scripts/Update_Students/createNewYml.js]: Trim all the values just in case

**Review by yangshun — CHANGES_REQUESTED** [review]

File: TA_Scripts/Update_Students/contactlist.csv (hunk 1 above)
Comment (Tzyinc) [reply TA_Scripts/Update_Students/contactlist.csv]: oh ya good point

File: TA_Scripts/Update_Students/dependancies/csv.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
+var fs = require('fs');
```
Comment (yangshun) [inline TA_Scripts/Update_Students/dependancies/csv.js:1]: Just use the csv to json npm module instead of vendoring it.

File: _data/students.yml [hunk 10]
```diff
@@ -1,4 +1,350 @@
 - batch:
+  year: 2017
+  students:
+  - name: Aaron Ong Chong Shi
+    id: aaron-ong-chong-shi
+    faculty: "FASS, Engineering, Business, Computing, SDE, Science"
```
Comment (ianngiaw) [inline _data/students.yml]: I think Aaron is Yale-NUS so he put all faculties. Shall we change this?

**Comment by louietyj** [comment]
I did some minor clean ups, let's merge this first so that it shows up on the website and then if we want we can come back and update the script.

**Pull request merged**